src/main/java/com/ibm/cloud/sdk/core/security/IamToken.java

```diff
@@ -119,11 +127,18 @@ public synchronized boolean needsRefresh() {
    * Check if the currently stored access token is valid. This is different from the needsRefresh method in that it
    * uses the actual TTL to calculate the expiration, rather than just a fraction.
    *
-   * @return true iff is the current access token is not expired
+   * @return true iff the current access token is valid and not expired
    */
   @Override
   public boolean isTokenValid() {
     return this.getException() == null
-        && (this.expiration == null || Clock.getCurrentTimeInSeconds() < this.expiration);
+        && (this.expiration == null || Clock.getCurrentTimeInSeconds() < (this.expiration - IamExpirationWindow));
+  }
+
+  public String toString() {
```

Comment:
This was useful during debugging, so I thought I'd leave it here.

Reply:
I was wondering if it's a good idea to automatically print the token by calling a single method, but I guess it's fine since the method belongs to the `IamToken` class so it should be obvious what it does.